middleware.ml

```diff
@@ -1,25 +1,63 @@
 type handler = Httpaf.Reqd.t -> unit Lwt.t
 type middleware = handler -> handler
 
-let has_session_cookie (reqd : Httpaf.Reqd.t) : bool =
+let has_session_cookie (reqd : Httpaf.Reqd.t) =
   let headers = (Httpaf.Reqd.request reqd).headers in
   match Httpaf.Headers.get headers "Cookie" with
   | Some cookies ->
       let cookie_list = String.split_on_char ';' cookies in
-      List.exists
+      List.find_opt
         (fun cookie ->
           let parts = String.trim cookie |> String.split_on_char '=' in
           match parts with
           | [ name; _ ] -> String.equal name "molly_session"
           | _ -> false)
         cookie_list
-  | _ -> false
+  | _ -> None
 
-let apply_middleware middlewares handler =
-  List.fold_right (fun middleware acc -> middleware acc) middlewares handler
+let apply_middleware ~now middlewares handler =
+  List.fold_right
+    (fun middleware acc -> middleware ~now acc)
+    middlewares handler
 
-let redirect_to_login reqd ?(msg = "") () =
-  let headers = Httpaf.Headers.of_list [ ("location", "/sign-in") ] in
+let get_expired_cookie ~now =
```

Comment:
I've to admit I don't quite understand why this function is around...

Reply:
You are correct, there's no reason to have this function.